pkg/server/image_status.go

```diff
@@ -17,48 +17,56 @@ limitations under the License.
 package server
 
 import (
-	"golang.org/x/net/context"
+	"fmt"
 
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
 	"k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+
+	"github.com/kubernetes-incubator/cri-containerd/pkg/metadata"
 )
 
 // ImageStatus returns the status of the image, returns nil if the image isn't present.
-func (c *criContainerdService) ImageStatus(ctx context.Context, r *runtime.ImageStatusRequest) (*runtime.ImageStatusResponse, error) {
-	ref := r.GetImage().GetImage()
-	// TODO(mikebrow): Get the id->tags, and id->digest mapping from our checkpoint;
-	// Get other information from containerd image/content store
-
-	// if the passed ref is a digest.. and is stored the following get should work
-	// note: get returns nil with no err
-	meta, _ := c.imageMetadataStore.Get(ref)
-	if meta != nil {
-		return &runtime.ImageStatusResponse{Image: &runtime.Image{ // TODO(mikebrow): write a ImageMetadata to runtim.Image converter
-			Id:          meta.ID,
-			RepoTags:    meta.RepoTags,
-			RepoDigests: meta.RepoDigests,
-			Size_:       meta.Size,
-			// TODO(mikebrow): Uid and Username?
-		}}, nil
+// TODO(random-liu): We should change CRI to distinguish image id and image spec.
+func (c *criContainerdService) ImageStatus(ctx context.Context, r *runtime.ImageStatusRequest) (retRes *runtime.ImageStatusResponse, retErr error) {
+	glog.V(4).Infof("ImageStatus for image %q", r.GetImage().GetImage())
+	defer func() {
+		if retErr == nil {
+			glog.V(4).Infof("ImageStatus for %q returns image status %+v",
+				r.GetImage().GetImage(), retRes.GetImage())
+		}
+	}()
+	imageID, err := c.localResolve(ctx, r.GetImage().GetImage())
+	if err != nil {
+		return nil, fmt.Errorf("can not resolve %q locally: %v", r.GetImage().GetImage(), err)
+	}
+	if imageID == "" {
+		// return empty without error when image not found.
```

Comment:
This is defined in CRI. https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/cri/v1alpha1/api.proto#L85-L88

And we're actually relying on this behavior.